microsoft_intune/icon_microsoft_intune/util/api.py

```diff
@@ -117,6 +133,30 @@ def _request(self, method, url, params=None, request_body=None, headers=None, da
             self.logger.info(f"Call to API failed: {e}")
             raise PluginException(preset=PluginException.Preset.UNKNOWN)
 
+    def get_device_by_uuid_if_not_whitelisted(self, device, whitelist):
+        device_response = self.search_managed_devices(device)
+
+        if not device_response:
+            raise PluginException(
+                cause=f"Managed device: {device}, was not found",
```

Comment:
These need punctuation, they're not added by SDK

```suggestion
                cause=f"Managed device: {device}, was not found.",
```